libminifi/test/sql-tests/mocks/MockConnectors.h

```diff
@@ -0,0 +1,175 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <regex>
+#include <map>
+#include <vector>
+#include <algorithm>
+#include <memory>
+#include <string>
+
+#include "data/DatabaseConnectors.h"
+#include "utils/StringUtils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace sql {
+
+class MockRow : public Row {
+ public:
+  MockRow(std::vector<std::string>& column_names, std::vector<DataType>& column_types, const std::vector<std::string>& column_values)
+    : column_names_(column_names), column_types_(column_types), column_values_(column_values) {
+  }
+
+  MockRow(MockRow&& other) = default;
+  MockRow(const MockRow& other) = default;
+  MockRow& operator=(MockRow&& other) {
+    column_names_ = other.column_names_;
+    column_types_ = other.column_types_;
```

Comment:
I don't know the exact intention here, but I wager it won't be doing that

Reply:
using `std::reference_wrapper` would most likely offer the semantics we are looking for here

Reply:
also won't the lack of `noexcept` make it unavailable for `std::vector`?

Reply:
You right this was wrong. The original goal was just to make the class copyable, and `std::reference_wrapper` could have been a good solution here. Finally I decided it's simpler just to use a raw pointer in this case. Updated in ee9452ebd591ff22139606ba2cd682df912d14f4

Reply:
I'd say just use pointers. Whether you write `*column_names_` or `column_names_.get()` doesn't really matter. Raw pointers are not evil as observer pointers.